### PR TITLE
feat(i18n): translate server messages

### LINKED ISSUE #40624: Translate server flash messages

### PR DESCRIPTION
**Pull request by moT01**
This should translate most of the commonly seen flash messages that are sent from the server to the client.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40624

### CODE REVIEW

**Comment by moT01** [comment]
I have opened this up for review. It should translate most of the commonly seen messages. Some of the other messages are nearly impossible to make their way to the client UI and may serve a better purpose staying in English.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Requesting @ojeytonwilliams for a second look as well.

File: api-server/server/boot/certificate.js [hunk 1]
```diff
@@ -496,9 +486,9 @@ function createShowCert(app) {
           messages: [
             {
               type: 'info',
-              message: dedent`
-              ${username} has not yet agreed to our Academic Honesty Pledge.
-            `
+              message: 'flash.msg-36',
```
Comment (RandellDawson) [inline api-server/server/boot/certificate.js:489]: FYI - I noticed that `msg-36` has a typo in the `translations.json` fiile (see [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/b8ce82c4f8c6c34b5855ed5b3fcc47f2efd2d78a/client/i18n/locales/english/translations.json#L432).

`agrees` should be `agreed`
```js
"msg-36": "{{username}} has not yet agreed to our Academic Honesty Pledge.",
```

File: api-server/server/boot/user.js [hunk 2]
```diff
@@ -246,8 +247,10 @@ function createPostReportUserProfile(app) {
         }
 
         return res.json({
-          typer: 'info',
-          message: `A report was sent to the team with ${user.email} in copy.`
```
Comment (RandellDawson) [inline api-server/server/boot/user.js:250]: Not that it makes much difference, but the `translation.json` message does not have a period at the end of the sentence like this does.

On a side note, I noticed a couple others that either added a period to the end that was not there or removed a period that was there in the original message.  Again, not a big deal, but after looking through all the messages (which is easier now that they are all in the same file), I notice we have not been consistent with using/not using periods at the end of the messages.  My personal preference would be to have a period if the message is a full sentence.

**Review by RandellDawson — APPROVED** [review]
I validated the corresponding messages match up to the original text.  I did not test out all the messages show locally, but the logic looks solid to me.

Great work @moT01!

File: client/src/components/Flash/index.js [hunk 3]
```diff
@@ -38,7 +44,7 @@ function Flash({ flashMessage, onClose }) {
             className='flash-message'
             onDismiss={handleClose}
           >
-            {needsTranslating ? t(`${message}`) : message}
+            {needsTranslating ? t(`${message}`, variables) : message}
```
Comment (ojeytonwilliams) [inline client/src/components/Flash/index.js:47]: Do we need this logic any more?  If `t` returns the key when there's no corresponding entry then we can just use 
```suggestion
           t(message, variables)
```
where I've also removed the coercion of `message` to a string.  As far as I can see it already is one.

**Review by ojeytonwilliams — COMMENTED** [review]
Nice work, Tom.  One general question: can we use more meaningful names for the message keys? Rather than `flash.msg-29` and the like.  I'd be happy to help with the naming process if we do (maybe in another PR if you'd prefer).

File: api-server/server/boot/user.js (hunk 2 above)
Comment (moT01) [reply api-server/server/boot/user.js:250]: I may have added and removed some periods here and there to try and make things consistent. But some messages/text seem to need them and some don't - it seems like a judgment call. Not sure if we should try and come up with a standard. I'm open to discussing it.

File: client/src/components/Flash/index.js (hunk 3 above)
Comment (moT01) [reply client/src/components/Flash/index.js]: Good call.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks Tom, this looks good.

**Comment by moT01** [comment]
Thanks for the reviews everyone 🎉 I spoke with Rafael, he is almost done translating the json file. I think we should wait until that gets added and then go and change up the names of those keys.

**Pull request merged**